python/cuml/common/array.py

```diff
@@ -0,0 +1,259 @@
+        if data is None:
+            raise TypeError("To create an empty Array, use the class method" +
+                            " Array.empty()")
+        if isinstance(data, int):
+            if dtype is None or shape is None or order is None:
+                raise TypeError("Need to specify dtype, shape and order when" +
+                                " creating an Array from a pointer.")
+
+        ary_interface = False
+
+        if isinstance(data, DeviceBuffer) or isinstance(data, int):
+            size, shape = _get_size_from_shape(shape, dtype)
+            super(Array, self).__init__(data=data, owner=owner, size=size)
+            self.shape = shape
+            self.dtype = np.dtype(dtype)
+            self.order = order
+            self.strides = _order_to_strides(order, shape, dtype)
+
+        elif hasattr(data, "__array_interface__"):
+            ary_interface = data.__array_interface__
+
+        elif hasattr(data, "__cuda_array_interface__"):
+            ary_interface = data.__cuda_array_interface__
+
+        else:
+            raise TypeError("Unrecognized data type.")
+
+        if ary_interface:
+            super(Array, self).__init__(data=data, owner=owner)
+            self.shape = ary_interface['shape']
+            self.dtype = np.dtype(data.dtype)
+            if ary_interface['strides'] is None:
+                self.order = 'C'
+                self.strides = _order_to_strides(self.order, self.shape,
+                                                 self.dtype)
+            else:
+                self.strides = ary_interface['strides']
+                self.order = _strides_to_order(self.strides, data.dtype)
+
+    def __getitem__(self, slice):
+        return Array(data=cp.asarray(self).__getitem__(slice))
+
+    def __setitem__(self, slice, value):
+        cp.asarray(self).__setitem__(slice, value)
+
+    def to_output(self, output_type='cupy'):
+        """
+        Convert array to output format
+
+        Parameters
+        ----------
+        output_type : string
+            Format to convert the array to. Acceptable formats are:
+            'cupy' - to cupy array
+            'numpy' - to numba array
+            'numba' - to numba device array
+            'dataframe' - to cuDF DataFrame
+            'series' - to cuDF Series
+            'cudf' - to cuDF Series if array in single dimensional, to
+                DataFrame otherwise
+        """
+
+        if output_type == 'cudf':
+            if len(self.shape) == 1:
+                output_type = 'series'
+            elif self.shape[0] > 1 and self.shape[1] > 1:
+                output_type = 'dataframe'
+            else:
+                output_type = 'series'
+
+        if output_type == 'cupy':
+            return cp.asarray(self)
+
+        elif output_type == 'numba':
+            return cuda.as_cuda_array(self)
+
+        elif output_type == 'numpy':
+            return np.array(cp.asnumpy(cp.asarray(self)), order=self.order)
```

Comment:
Would it make sense to just use [`cp.asnumpy`]( https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.asnumpy.html#cupy.asnumpy )'s `order` argument?